tslint.ts

```diff
@@ -1,10 +1,74 @@
+import * as fs from 'fs';
+import * as path from 'path';
+import * as coreModules from 'resolve/lib/core';
 import * as tslint from 'tslint';
 import * as tslintAirbnb from 'tslint-config-airbnb';
 const tslintReact = require('tslint-react');
 
 const rulesAirbnb = tslintAirbnb.rules;
 const rulesReact = tslintReact.rules;
 
+// prepare regex for node core modules for 'ordered-imports' rule
+const modules = Object.entries(coreModules as Record<string, boolean>).reduce(
+  (acc: string[], [module, enabled]) => {
+    enabled && !module.startsWith('_') && acc.push(module);
+    return acc;
+  },
+  []
+);
+const coreModulesRegex = '^(' + modules.join('|') + ')$';
+
+// prepare import groups for 'ordered-imports' rule
+const projectPackageJson = path.join(process.cwd(), 'package.json');
+let dependenciesRegex;
+if (fs.existsSync(projectPackageJson)) {
+  const packageJson = require(projectPackageJson);
+  const dependencies = [
+    ...Object.keys(packageJson.dependencies || {}),
+    ...Object.keys(packageJson.devDependencies || {}),
+  ];
+
+  if (dependencies.length) {
+    dependenciesRegex = '^(' + dependencies.join('|').replace(/\./g, '\\\\.') + ')(/|$)';
+  }
+}
+
+const importGroups = [
+  {
+    name: 'aliased paths, which begin with tilde in our convention',
+    match: '^~',
+    order: 30,
+  },
+  {
+    name: 'parent directories',
```

Comment:
will the "farthest" imports be put on top?

Reply:
the import groups are matched first, then ordered by `order` property

Reply:
ah you meant the farthest as in the number of `../` - the imports will be ordered alphabetically, so most probably the farthest will be on the top within the group